python/tests/operations/join.py

```diff
@@ -275,6 +275,55 @@ def test_different_names(self):
 
         self.assertTrue(np.equal(joined_labels.tolist(), ref).all())
 
+    def test_different_names_different_length(self):
+        """Test Label joining using labels with different names and different length."""
+        property_labels_1 = Labels(
+            ("structure", "prop_1"), np.vstack(2 * [np.arange(5)]).T
+        )
+        property_labels_2 = Labels(
+            ("structure", "prop_2", "prop_3"), np.vstack(3 * [np.arange(5)]).T
+        )
+
+        block_1 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_1,
+        )
+
+        block_2 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_2,
+        )
+
+        joined_ts = eop.join(
+            [TensorMap(self.keys, [block_1]), TensorMap(self.keys, [block_2])],
+            axis="properties",
+        )
+
+        joined_labels = joined_ts.block(0).properties
+
+        self.assertEqual(joined_labels.names, ("tensor", "property"))
```

Comment:
where are these names coming from? I don't see them anywhere defined. The property labels before are ("structure", "prop_1") and ("structure", "prop_2", "prop_3"). I really have problems to understand the join function. Can we improve the doc of it.

Reply:
Yes it is quite complicated. The problem is joining meta data that is not unique. Thus problem is exactly what the `join_labels` function is trying to solve. I  move the docstring into the join function.

Reply:
thanks the doc helps a lot, put some comments there